backend/services/operations.go

```diff
@@ -417,11 +422,12 @@ func SetFavoriteOperation(ctx context.Context, db *database.Connection, i SetFav
 		return backend.WrapError("Unwilling to read operation", backend.UnauthorizedReadErr(err))
 	}
 
-	err = db.Update(sq.Update("user_operation_permissions").
-		SetMap(map[string]interface{}{
-			"is_favorite": i.IsFavorite,
-		}).
-		Where(sq.Eq{"operation_id": operation.ID, "user_id": middleware.UserID(ctx)}))
+	_, err = db.Insert("user_operation_preferences", map[string]interface{}{
+		"user_id": middleware.UserID(ctx),
+		"operation_id": operation.ID,
+		"is_favorite": i.IsFavorite,
+	}, "ON DUPLICATE KEY UPDATE is_favorite=VALUES(is_favorite)")
```

Comment:
Is that how you do an upsert? For my current PR I was trying to figure out if that was possible with the squirrel (it's not a supported method), but this might solve the problem for me!

Reply:
good to know!